internal/page/dashboard.go

```diff
@@ -56,6 +47,20 @@ func Dashboard(tmpl template.Template, donorStore DonorStore, dashboardStore Das
 			tabCount++
 		}
 
+		if r.Method == http.MethodPost {
+			lpa, err := donorStore.Create(r.Context())
+			if err != nil {
+				return err
+			}
+
+			path := Paths.YourDetails
+			if len(donorLpas) > 0 {
```

Comment:
Alternatively, could add a hidden field to the template if donorLpas>0 then check that on POST. That would save doing the dashboardStore.GetAll query when making a new one.